src/ui/panels/clan.js

```diff
@@ -99,20 +103,36 @@ export class Clan {
     }
 
     // Add clan member to list
-    addClanMember(member) {
+    addClanMember(memberName) {
         // Add to saved data
-        characterData.addClanMember(member.name);
+        characterData.addClanMember(memberName);
 
         // Add visuals if the dashboard is up
         if (this.dashboard.scene.isActive()) {
             const startX = 560,
                 startY = 280,
                 yDiff = 18;
 
-            // Add text to the list
-            let index = this.clanMembers.length;
-            member.createText(false, startX, startY + index * yDiff);
-            this.clanMembers.push(member);
+            // Add new clan entry if new member else increment existing member by one
+            const index = this.clanMembers.findIndex((member) => {
+                if (member.name === memberName) {
+                    return true;
+                }
+                return false;
+            });
+
+            if (index != -1) {
+                this.clanMembers[index].numberOwned++;
+            } else {
```

Comment:
It's not necessary, but if you use Curtis's revision, you could inverse the ```if (existingMember)``` condition to remove the else statement and two curly braces to make the code a little more readable. You would need to assign the existingMember variable to the newMember if ```(!existingMember)``` is true since we would always be incrementing existingMember in this case.

Example:

```
let existingMember = this.clanMembers.find((member) => member.name === memberName);

if (!existingMember) {
  const newMember = getAutoclickerClass(memberName, this.scrollWindow);
  newMember.createText(
                  false,
                  startX,
                  startY + this.clanMembers.length * yDiff
              );
  newMember.numberOwned = 0;
  this.clanMembers.push(newMember);
  existingMember = newMember;
} // No else statement needed!!!

existingMember.numberOwned++;

Reply:
Kind of, but they're a little different. Both maps and objects use key and value pairs to store data.

The one big difference that might be useful are that maps keep the order of elements preserved, while objects do not.

I also prefer the syntax of maps over objects in this situation. For example to determine if a map has a key, you can just do '''map.has(key)''', while for object the equivalent would be '''object.key === undefined'''. Another example is '''map.size''' vs '''Object.keys(object).length''' to get the number of items.

We should use object though, since JSON supports objects and not maps. It sounds like there may be future support for maps in the future though.

I was going to put an article showcasing the differences but it's a pain to post links on my phone.

Reply:
Oh gotcha, I guess I never realized they had separate object/dictionaries and an actual Map class. Good to know

Reply:
https://github.com/EthanVieira/rs-clicker/pull/71/commits/8e37d85adae8b81c4d9a1d9b3e40c7f746834278